### PR TITLE
Add doc strings for MentorshipRelationDAO class and its functions.

### LINKED ISSUE #155: Create doc strings for MentorshipRelationDAO class and its functions
## Description
As a developer,
I need to have in code documentation,
so that I can understand better the code.

Related to #149 

We need to have **doc strings for MentorshipRelationDAO class and its functions**. You can find this file [app/api/dao/mentorship_relation.py](https://github.com/systers/mentorship-backend/blob/develop/app/api/dao/mentorship_relation.py).
You can follow these conventions: [style guide for doc strings in python by Google](https://github.com/google/styleguide/blob/gh-pages/pyguide.md#38-comments-and-docstrings).

### Update [Required]
- [ ] Create doc strings for MentorshipRelationDAO class and its functions (file: app/api/dao/mentorship_relation.py).

## Definition of Done
- [ ] All of the required items are completed.
- [x] Approval by 1 mentor.

## Estimation
1 hour

### PR DESCRIPTION
**Pull request by agarwalrounak**
### Description

Added doc strings for MentorshipRelationDAO class and its functions.

Fixes #155 

### Type of Change:

- Documentation

### How Has This Been Tested?

No tests required.

### Checklist:

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas

### CODE REVIEW

File: app/api/dao/mentorship_relation.py
Comment (isabelcosta) [inline app/api/dao/mentorship_relation.py]:
```suggestion
    """Data Access Object for mentorship relation functionalities.
```

File: app/api/dao/mentorship_relation.py
Comment (isabelcosta) [inline app/api/dao/mentorship_relation.py:349]: add line above

File: app/api/dao/mentorship_relation.py
Comment (isabelcosta) [inline app/api/dao/mentorship_relation.py]:
```suggestion
        """Lists mentorship requests in pending state.
```

File: app/api/dao/mentorship_relation.py
Comment (isabelcosta) [inline app/api/dao/mentorship_relation.py]:
```suggestion
        """Lists current mentorship relation details.
```

File: app/api/dao/mentorship_relation.py
Comment (isabelcosta) [inline app/api/dao/mentorship_relation.py]:
```suggestion
        """Lists past mentorship relations.
```

File: app/api/dao/mentorship_relation.py
Comment (isabelcosta) [inline app/api/dao/mentorship_relation.py]:
```suggestion
        """Deletes a mentorship request.
```

File: app/api/dao/mentorship_relation.py
Comment (isabelcosta) [inline app/api/dao/mentorship_relation.py]:
```suggestion
        Deletes a mentorship request if the current user was the one who created it and the request is in the pending state.
```

File: app/api/dao/mentorship_relation.py
Comment (isabelcosta) [inline app/api/dao/mentorship_relation.py]:
```suggestion
        """Rejects a mentorship request.
```

Add a longer description 1 line below:
```
Reject mentorship request if the current user if is the receiver of the request.
```

File: app/api/dao/mentorship_relation.py
Comment (isabelcosta) [inline app/api/dao/mentorship_relation.py]:
```suggestion
        """Allows a mentorship request.
```

**Review by isabelcosta — CHANGES_REQUESTED** [review]
@agarwalrounak you're doing great!
I just left some comments for you to fix on the doc strings. Once you do that I can approve and merge this ;)

**Comment by agarwalrounak** [comment]
Thanks for the review @isabelcosta :smiley: 
I have addressed the requested changes.

**Comment by agarwalrounak** [comment]
@isabelcosta I have addressed your change requests. Please review :)

**Review by isabelcosta — DISMISSED** [review]
Good work @agarwalrounak 🎉

**Comment by isabelcosta** [comment]
@agarwalrounak I'll merge this in the next days. I will wait for other PRs if there's any so that I can merge more than one at a time. This is because the Database is volatile, whenever we merge something, all data will be deleted. So data related with people working on Android app, will be reset, and I'll have to create the admin user.

**Review by ramitsawhney27 — CHANGES_REQUESTED** [review]
The message for what each of the methods return is not descriptive enough. It's the same generic thing, it should be more verbose.

**Comment by agarwalrounak** [comment]
Thanks for the review @ramitsawhney27 

I had two ideas for how the messages could be written:
1) A message corresponding to the completed action; success if mentorship relationship is established, failure if otherwise.
2) A success message if mentorship relationship is established, failure message if otherwise.

Which of the above two do you think would be more appropriate? Or any other way the message could be written?

**Review by ramitsawhney27 — COMMENTED** [review]
The first seems fine

**Comment by agarwalrounak** [comment]
I have addressed the change requests and updated the PR. 
@ramitsawhney27 Please review :)

**Comment by agarwalrounak** [comment]
@isabelcosta changes have been applied. Let me know if anything is wrong.

**Review by ramitsawhney27 — APPROVED** [review]
Great work, thanks for contributing!

**Review by isabelcosta — APPROVED** [review]
Good job @agarwalrounak

**Pull request merged**
